python/planout/ops/core.py

```diff
@@ -90,6 +91,15 @@ def pretty(self):
         return f
 
 
+class Map(PlanOutOpSimple):
+
+    def simpleExecute(self):
+        m = deepcopy(self.args)
+        del m['op']
+        del m['salt']
```

Comment:
Yup --- the current implementation of PlanOut adds salt to the RHS of any
"set" expression, regardless of whether the operator is a random op.  We
could modify that if statement to say something like

```
if ops.Operators.isOperator(value) and 'salt' not in value and issubclass(
Operators.operators[value.op], PlanOutOpRandom)
```

If doing the issubclass operation didn't seem like it would be a lot slower
than deleting a key from a dictionary (which could depend on the language
PlanOut is implemented in), I would definitely adopt the above, but would
want to do some experimentation first...


e

On Tue, Jun 23, 2015 at 1:25 PM, Sujeeth <notifications@github.com> wrote:

> In python/planout/ops/core.py
> <https://github.com/facebook/planout/pull/68#discussion_r33087604>:
>
> > @@ -90,6 +91,15 @@ def pretty(self):
> >          return f
> >
> >
> > +class Map(PlanOutOpSimple):
> > +
> > +    def simpleExecute(self):
> > +        m = deepcopy(self.args)
> > +        del m['op']
> > +        del m['salt']
>
> args['salt'] would only be present for random ops. Right?
>
> Or perhaps it is coming from here(?)
>
>         # if the value is operator, add the name of the variable as a salt if no
>         # salt is provided.
>         if ops.Operators.isOperator(value) and 'salt' not in value:
>             value['salt'] = var
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/planout/pull/68/files#r33087604>.
>